README.md

```diff
@@ -27,25 +117,148 @@ mtx.myTranspose();
+let arr = [1,1,1,1,4];
+console.log(arr.myReduce((sum, el) => { return sum+el; }))
+```
+12. Use `myMap` to build a new array that doubles each element. Then chain `myReduce` to find the product of all the elements.
+```js
+let arr = [1,1,1,1,4];
+let doubleProductArr = arr.myMap( el => { return el*2; }).myReduce((acc, num)) => { return acc*num; }
+console.log(doubleProductArr)
+
+```
+13. Use `myReduce` to find the largest number in an array.
+```js
+let arr = [1,1,1,1,4];
+let doubleProduct = arr
+    .map(num => {return num*2; })
+    .reduce((total,num) => {return total*num; })
+console.log(doubleProduct)
+```
 14. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.
 Use `myFilter` to return a new array of every element of arr that can be evenly divided by divisor.
-14. Write a `myJoin` function. 
+```js
+const elementDivisibleBy = (divisor,arr) => {
+  return arr.myFilter(el => {return el%divisor===0; })
+}
+console.log(elementDivisibleBy(2,[1,3,6,5]))
+```
+14. Write a `myJoin` function.
+(disclaimer: it works but it happened by mistake. I don't understand fully why this works)
+Function:
+```js
+const myJoin = (arr,s=',') => {
+  return arr.reduce((acc,el) => {return acc+s+el.toString();})
+}
+console.log(myJoin(['w','t','h'],','))
+```
+
+Method:
+```js
+Array.prototype.myJoin = function(s) {
+  return this.reduce((acc,el) => {return acc+s+el.toString();});
+}
+let arr = ['w','t','h'];
+let joined = arr.myJoin('');
+console.log(joined)
+```
 15. Write a `mySlice` function.
+```js
+Array.prototype.mySlice = function(i,j) {
+  return this.filter( el => {
+    return (this.indexOf(el)<this.indexOf(this[j])
+            && this.indexOf(el)>=this.indexOf(this[i]));
+  });
+}
+let arr = ['b','u','l','z','i'];
+let ulz = arr.mySlice(1,4);
+console.log(ulz)
+```
 16. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
 Use reduce.
-17. Write a `numberTimesIdx` that uses `map` and multiples each number in the array by it's index. 
 
-Bonus: Write a `myFlatten`. This should take a multi-dimensional array and return it as one array. 
+I had to use here Initial value=0, and not sure why-please explain!
+```js
+const countZeroes = (arr) => {
+  return arr.reduce((totalC,el) => {
+  if (el===0) {
+    totalC+=1;
+  }
+  return totalC;
+},0);
+}
+console.log(countZeroes([0,0,0,0,4,3]))
+```
+17. Write a `numberTimesIdx` that uses `map` and multiples each number in the array by it's index.
+```js
+const numberTimesIdx = (arr) => {
+  return arr.map(el => {return el*arr.indexOf(el)});
+}
+let arr = [1,2,3,4,5];
+console.log(numberTimesIdx(arr))
+```
+Bonus: Write a `myFlatten`. This should take a multi-dimensional array and return it as one array.
 ```js
 let arr = [1, 2, [3, 4, 5, [6, 7, 8]]]
   arr.myFlatten();
   // => [1, 2, 3, 4, 5, 6, 7, 8]
 ```
+My best attempt is cutting close but doesn't work:
+
+```js
+let arr = [1, 2, [3, 4, 5, [6, 7, 8]]]
+
```

Comment:
This is very close! You are missing using recursion which we haven't discussed yet. Great attempt!